controller/controlplane/status_conditions.go

```diff
@@ -6,13 +6,12 @@ import (
 	k8sutils "github.com/kong/gateway-operator/pkg/utils/kubernetes"
 
 	kcfgcontrolplane "github.com/kong/kubernetes-configuration/api/gateway-operator/controlplane"
-	operatorv1beta1 "github.com/kong/kubernetes-configuration/api/gateway-operator/v1beta1"
 )
 
 // markAsProvisioned marks the provided resource as ready by the means of Provisioned
 // Status Condition.
-func markAsProvisioned[T *operatorv1beta1.ControlPlane](resource T) {
-	cp, ok := any(resource).(*operatorv1beta1.ControlPlane)
+func markAsProvisioned[T *ControlPlane](resource T) {
```

Comment:
It results in

```
  Conditions:
    Last Transition Time:  2025-06-13T09:16:04Z
    Message:               
    Observed Generation:   1
    Reason:                Ready
    Status:                True
    Type:                  Ready
    Last Transition Time:  2025-06-13T09:16:04Z
    Message:               pods for all Deployments are ready
    Observed Generation:   1
    Reason:                PodsReady
    Status:                True
    Type:                  Provisioned
```

There used to be `Pod`s for `ControlPlane`, but it's not true anymore. Perhaps that message/conditions should be adjusted. It don't need to be done in this PR, I'll put it more just to have a discussion

Reply:
Done in 5b92888507719b2ca5d5d581921b1158fcd56c56.